lib/install/validate/validator.go

```diff
@@ -501,21 +504,147 @@ func (v *Validator) certificateAuthorities(ctx context.Context, input *data.Data
 
+	// Check if CAs can be loaded
+	pool := x509.NewCertPool()
+	if len(input.RegistryCAs) > 0 {
+		if !pool.AppendCertsFromPEM(input.RegistryCAs) {
+			v.NoteIssue(errors.New("Unable to load certificate authority data for registry"))
+			return
+		}
+	}
+
+	conf.RegistryCertificateAuthorities = input.RegistryCAs
+
+	// test reachability
+	insecureRegistries, whitelistRegistries, err := v.reachableRegistries(ctx, input, pool)
+	if err != nil {
+		v.NoteIssue(err)
+		return
+	}
+
 	// copy the list of insecure registries
-	conf.InsecureRegistries = input.InsecureRegistries
+	conf.InsecureRegistries = insecureRegistries
+
+	// copy the list of whitelist registries
+	conf.RegistryWhitelist = whitelistRegistries
+
+	// create vic-machine info message
+	msg := v.friendlyRegistryList("Insecure registries", conf.InsecureRegistries)
+	if msg != "" {
+		log.Infof(msg)
+	}
+	msg = v.friendlyRegistryList("Whitelist registries", conf.RegistryWhitelist)
+	if msg != "" {
+		log.Infof(msg)
+	}
 
 	if len(input.RegistryCAs) == 0 {
 		return
 	}
+}
 
-	// Check if CAs can be loaded
-	pool := x509.NewCertPool()
-	if !pool.AppendCertsFromPEM(input.RegistryCAs) {
-		v.NoteIssue(errors.New("Unable to load certificate authority data for registry"))
-		return
+func (v *Validator) friendlyRegistryList(registryType string, registryList []url.URL) string {
+	var msg string
+
+	if len(registryList) > 0 {
+		var host string
+		msg = registryType + " = "
+		for i, r := range registryList {
+			host = r.Host
+			if strings.Contains(r.Path, "/") {
+				host = host + r.Path
+			}
+
+			if i == 0 {
+				msg = msg + host
+			} else {
+				msg = msg + ", " + host
+			}
+		}
 	}
 
-	conf.RegistryCertificateAuthorities = input.RegistryCAs
+	return msg
+}
+
+// Validate registries are reachable.  Secure registries that are not specified as insecure are validated with the
+// CA certs passed into vic-machine.
+func (v *Validator) reachableRegistries(ctx context.Context, input *data.Data, pool *x509.CertPool) (insecureRegistries []url.URL, whitelistRegistries []url.URL, err error) {
+	secureRegistries := input.WhitelistRegistries
+
+	// Remove intersection between insecure registries and whitelist registries from whitelist set so
+	// we can ensure we test the exclusion set with certs
+	var exist bool
+	var idx int
+	var s url.URL
+	for _, r := range input.InsecureRegistries {
+		exist = false
+		for idx, s = range secureRegistries {
+			if r.Host == s.Host {
+				exist = true
+				break
+			}
+		}
+
+		// remove the insecure registry from list of registries to get validated against certs
+		if exist {
+			secureRegistries = append(secureRegistries[:idx], secureRegistries[idx+1:]...)
+		}
+	}
+
+	// Test insecure registries' reachability
+	for _, r := range input.InsecureRegistries {
+		// Make sure address is not a wildcard domain or CIDR.  If it is, do not validate.
+		if strings.HasPrefix(r.Host, "*") {
+			log.Infof("Skipping registry validation for %s", r.Host)
```

Comment:
Make this debug - we should tell the user if a check fails, we don't tell them about success or skipping of other checks (or we shouldn't be).
We should also explain why we're skipping the check when it's in debug.